### ISSUE TITLE
[CI] SearchHitTests.testSerializeShardTarget fails on 6.x

### ISSUE THREAD

**Issue by tlrx** [elastic/elasticsearch#37214]
After #36997, the `SearchHitTests.testSerializeShardTarget` needs to be adapted to work on 6.x when a random version < 5.6 is picked up and the cluster alias is not deserialized.

See:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+release-tests/317/consoleFull

I can reproduce on 6.x with:
```
./gradlew :server:unitTest -Dtests.seed=6E493E03085BDF45 -Dtests.class=org.elasticsearch.search.SearchHitTests -Dtests.method="testSerializeShardTarget" -Dtests.security.manager=true -Dbuild.snapshot=true -Dtests.jvm.argline="-Dbuild.snapshot=true" -Dtests.locale=sr-BA -Dtests.timezone=Australia/ACT -Dcompiler.java=11 -Druntime.java=8
```

**Comment by tlrx** [comment]
Test muted on 6.x in https://github.com/elastic/elasticsearch/commit/3bca3d4a24b47bd9c4093cf0bb42863d29867748

**Comment by javanna** [comment]
Fixed in 890fa3453c157dab854f9e4e515b52c7098d7ba5
